src/pk/ecc/ecc_sign_hash.c

```diff
@@ -103,13 +139,29 @@ static int _ecc_sign_hash(const unsigned char *in,  unsigned long inlen,
       *outlen = 2*pbytes;
       err = CRYPT_OK;
    }
+   else if (sigformat == LTC_ECCSIG_ETH27) {
+      /* Ethereum (v,r,s) format */
+      if (key->dp.oidlen != 5   || key->dp.oid[0] != 1 || key->dp.oid[1] != 3 ||
+          key->dp.oid[2] != 132 || key->dp.oid[3] != 0 || key->dp.oid[4] != 10) {
+         /* Only valid for secp256k1 - OID 1.3.132.0.10 */
+         err = CRYPT_ERROR; goto errnokey;
+      }
+      if (*outlen < 65) { err = CRYPT_MEM; goto errnokey; }
+      zeromem(out, 65);
+      i = mp_unsigned_bin_size(r);
+      if ((err = mp_to_unsigned_bin(r, out + 32 - i)) != CRYPT_OK) { goto errnokey; }
+      i = mp_unsigned_bin_size(s);
+      if ((err = mp_to_unsigned_bin(s, out + 64 - i)) != CRYPT_OK) { goto errnokey; }
+      out[64] = (unsigned char)(v + 27); /* Recovery ID is 27/28 for Ethereum */
```

Comment:
before this line we should perhaps check whether `(v + 27) < 256`

Reply:
I guess that's up to the specification that defines that format and I can also very well imagine that `v` could be okay to overflow in this case as it's "the parity" of the key

Reply:
The only way it could overflow is for a curve with a cofactor of at least 114, which will break other things as then the order is not the same (bit-)length as the prime modulus. None of the defined ECC curves have a cofactor >4, which means the largest value for `v` will be 9.

Reply:
`LTC_ECCSIG_ETH27` is now restricted to only the secp256k1 curve, so 0<=v<4, with values of 2/3 being effectively impossible (probability less than 1-in-2^127), meaning no overflow is possible.